open-sphere-plugins/my-places/src/main/java/io/opensphere/myplaces/controllers/RoiContextMenuProvider.java

```diff
@@ -42,13 +42,16 @@ public RoiContextMenuProvider(Toolbox toolbox)
             MyPlacesDataTypeInfo type = (MyPlacesDataTypeInfo)key.getDataType();
             Placemark kmlPlacemark = type.getKmlPlacemark();
 
-            if (!(kmlPlacemark.getGeometry() instanceof Polygon))
+            if (kmlPlacemark.getGeometry() instanceof Polygon)
+            {
+                PolygonGeometry geom = RegionUtils.createGeometry(kmlPlacemark);
+                menuItems = MantleToolboxUtils.getMantleToolbox(myToolbox).getSelectionHandler().getGeometryMenuItems(geom);
+            }
+            else
```

Comment:
Do we need this else anymore with your changes?

Reply:
Yeah, this is to distinguish the menu items if its a track vs point vs line . The previous !(instance of ) was not catching a few cases , so this just makes it easier

Reply:
Its a little unrelated to my ticket, but I thought Id just fix it while I was getting the error testing this ticket

Reply:
no we dont , good catch